### PR TITLE
feat(client): web social sign in

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

implement web social sign in callback and error handler.

## Linear Issue Reference

LOG-1514

### Normal flow

1. Click "social" then open GitHub auth session
image
image

2. Sign in, then successfully get code
image

### Error handling
1. Click "social" then open GitHub auth session
2. Directly close the session w/o signing in, shows a error toast

image

## UI Code Reference

<details>
<summary>Click to expand</summary>

```tsx
// Sign in component
useEffect(() => {
  window.addEventListener('message', function (event) {
    if (event.origin === location.origin) {
      setToast(JSON.stringify(event.data));
    }
  });
}, []);

// `redirectTo` is the result of `POST /api/session/sign-in/social`
<Button
  onClick={() => {
    const { socialHandler } = window.webkit?.messageHandlers ?? {};
    socialHandler?.postMessage({
      callbackUri: 'http://localhost:3001/foo', // this should be the same as normal callback url when implement
      redirectTo:
        'https://github.com/login/oauth/authorize?client_id=...',
    });
  }}
>
  social
</Button>

// Route components
const GitHub = () => {
  useEffect(() => {
    window.location.assign(`logto-callback://github${window.location.search}`);
  }, []);

  return <div>loading</div>;
};

const Foo = () => {
  return <div>success: {window.location.href}</div>;
};

// Added routes
<Route exact path="/callback/github" component={GitHub} />
<Route exact path="/foo" component={Foo} /> // should merge into `/callback`
```

</details>

## Reviewers

@logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift [hunk 1]
```diff
@@ -25,9 +26,80 @@ public class LogtoWebViewAuthViewController: UnifiedViewController {
     override public func loadView() {
         view = webView
         webView.navigationDelegate = authSession
+        webView.configuration.userContentController.add(self, name: "socialHandler")
```
Comment (xiaoyijun) [inline Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift:29]: `socialHandler` needs to align with both Android and iOS platforms.
Maybe we need an issue to track this.

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift (hunk 1 above)
Comment (gao-sun) [reply Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift:29]: the idea is great, but `window.webkit` is more like an iOS stuff. not sure if we can align

Comment (gao-sun) [reply Sources/LogtoClient/LogtoAuthSession/LogtoWebViewAuthViewController.swift:29]: added LOG-2053, let's figure out this later

**Pull request merged**